src/config.rs

```diff
@@ -139,6 +151,17 @@ impl FlokiConfig {
 
         Ok(config)
     }
+
+    /// Replace all instances of ${user_env:VAR} in the yaml with the value of the local environmental variable "VAR".
```

Comment:
We probably want to reserve `${<mode>:<value>}` completely - so if we see `-e ${lol:foo}` we might want to throw an error rather than leaving it un-changed (this also allows adding additional things later e.g. `-e KERNEL_VERSION=${exec:uname -r}` without being a breaking change each time).

This also means that `${user_env:@*&}` wants to be an error too - erroring when someone it looks like someone has tried but failed to use a feature is probably better than assuming they weren't aware of the feature and meant to type what they typed (might require an escape hatch for the incredibly rare case where someone _wanted_ the un-expanded value)